### PR TITLE
[Material] Remove opacity from dark theme overlay check

### PR DESCRIPTION
**Pull request by johnsonmh**
## Description

This PR removes the opacities from the color check in the elevation overlay. This fixes the bug that `Material`s with `ColorScheme.surface.withOpacity(.96)` do not get the elevation overlay, even though they should. 

## Tests

I added the following tests:

* Material in a dark theme app with a `surface` color with 75% opacity still receives the dark theme overlay.

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/material/elevation_overlay.dart [hunk 1]
```diff
@@ -49,7 +49,7 @@ class ElevationOverlay {
     if (elevation > 0.0 &&
         theme.applyElevationOverlayColor &&
         theme.brightness == Brightness.dark &&
-        color == theme.colorScheme.surface) {
+        color.withOpacity(1.0) == theme.colorScheme.surface.withOpacity(1.0)) {
```
Comment (darrenaustin) [inline packages/flutter/lib/src/material/elevation_overlay.dart:52]: This test is getting more and more complicated. Seems like it is just a band-aid over the fact that we really don't have a concept of what is a surface and what isn't. That said, this is a good fix for now. Thx.

File: packages/flutter/test/material/material_test.dart [hunk 2]
```diff
@@ -343,6 +343,27 @@ void main() {
       expect(model.color, equals(Colors.cyan));
     });
 
+    testWidgets('overlay will apply to materials with a non-opaque surface color', (WidgetTester tester) async {
```
Comment (darrenaustin) [inline packages/flutter/test/material/material_test.dart:346]: Nice test.

**Review by darrenaustin — APPROVED** [review]
LGTM.

File: packages/flutter/lib/src/material/elevation_overlay.dart (hunk 1 above)
Comment (johnsonmh) [reply packages/flutter/lib/src/material/elevation_overlay.dart:52]: Agreed, we are kind of patching things up. Material-Android has gotten to a stable place with dark theme logic, so hopefully once we've patched the same holes that they have, we will be able to leave it alone. Removing opacities is another part of their check.

File: packages/flutter/test/material/material_test.dart (hunk 2 above)
Comment (johnsonmh) [reply packages/flutter/test/material/material_test.dart:346]: Thanks! 👍

**Review by perclasson — APPROVED** [review]
LGTM

**Review by JoseAlba — APPROVED** [review]
```
██╗      ██████╗ ████████╗███╗   ███╗
██║     ██╔════╝ ╚══██╔══╝████╗ ████║
██║     ██║  ███╗   ██║   ██╔████╔██║
██║     ██║   ██║   ██║   ██║╚██╔╝██║
███████╗╚██████╔╝   ██║   ██║ ╚═╝ ██║
╚══════╝ ╚═════╝    ╚═╝   ╚═╝     ╚═╝
```

**Pull request merged**